File: java/templates/api_extra/message_create_body.java
MessageInInternal msgInInternal = new MessageInInternal(messageIn);
if (msgInInternal.transformationsParams != null) {
    if (msgInInternal.transformationsParams.get("rawPayload") == null) {
        // transformationsParams may be immutable
        HashMap<String, Object> trParams =
                new HashMap<>(msgInInternal.transformationsParams);
        trParams.put("rawPayload", msgInInternal.payload);
        msgInInternal.transformationsParams = trParams;
    }
} else {
    HashMap<String, Object> trParam = new HashMap<>();
    trParam.put("rawPayload", msgInInternal.payload);
    msgInInternal.transformationsParams = trParam;
}
msgInInternal.payload = new HashMap<>();

File: kotlin/templates/api_extra/message_create_body.kt
var msgInInternal =
    MessageInInternal(
        messageIn.application,
        messageIn.channels,
        messageIn.eventId,
        messageIn.eventType,
        mapOf(),
        messageIn.payloadRetentionHours,
        messageIn.payloadRetentionPeriod,
        messageIn.tags,
        messageIn.transformationsParams,
    )
if (msgInInternal.transformationsParams != null) {
    // only set rawPayload if not already set
    if (msgInInternal.transformationsParams!!["rawPayload"] == null) {
        var trParams =
            (msgInInternal.transformationsParams as Map<String, Any>).toMutableMap()
        trParams["rawPayload"] = messageIn.payload
        msgInInternal.transformationsParams = trParams.toMap()
    }
} else {
    val trParams = mapOf("rawPayload" to messageIn.payload)
    msgInInternal.transformationsParams = trParams
}
